doc/api-extensions.md

```diff
@@ -2578,3 +2578,11 @@ The following new configuration options are introduced:
 ## `oidc_scopes`
 
 This introduces a new `oidc.scopes` server configuration key which can take a comma separate list of OIDC scopes to request from the identity provider.
+
+## `startup_scriptlet`
+
+This adds the ability to run scriptlets at various stages of startup:
+
+* `startup.scriptlet.early`
+* `startup.scriptlet.pre-start`
+* `startup.scriptlet.post-start`
```

Comment:
I still think this would be better handled by a single `raw.qemu.scriptlet` which gets passed the hook name as an argument.

The reason to avoid a generic `startup.scriptlet` type infrastructure is that we need to think about security. QMP calls are inherently unsafe so this must be a very restricted key that requires a trusted project, so having that particular scriptlet be under `raw.qemu` with the rest of the low-level qemu keys makes sense.

Reply:
Got it! I understand your remark, but because it is very dangerous to inject QMP commands, I thought that we might as well allow everything in a generic `startup.scriptlet`.

Before I do the proper renamings, are you talking about passing the hook name as an argument to the Starlark function and letting the user deal with it with an `if`/`elif`/`else` construct? What would you want the function to be called if `startup` isn’t suited?

The changes needed are pretty light, I’ll try to do them today.